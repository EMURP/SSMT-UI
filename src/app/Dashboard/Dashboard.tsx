import React from 'react';
import { PageSection, Title, Stack, StackItem } from '@patternfly/react-core';

//import { ProjectDataFilterForm } from '@app/DataFilterForm/ProjectDataFilterForm';
import { DemoProjectDataFilterForm } from '@app/project_page/demoProjectfilterform';

// import  {ProjectInfoTableComponent}  from '../ProjectInfo/ProjectInfoTableComponent';


type myProps = {};
type myState = {
    startHrs: number;
    endHrs: number;
    startDate: Date;
    endDate: Date;
};

const convertDateToUTC = (date: Date )=> { 
    return new Date(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate(),
     date.getUTCHours(), date.getUTCMinutes(), date.getUTCSeconds()); }

class Dashboard extends React.Component<myProps, myState> {
  constructor(myProps) {
    super(myProps);
    this.state={
        startHrs: 0,
        endHrs: 0,
        startDate: new Date(Date.UTC(0, 0, 0, 0, 0, 0)),
        endDate: convertDateToUTC(new Date()),
    }
  }

  render() {
    return (
      <PageSection>
        <Title headingLevel="h1" size="lg">
          Dashboard Page Title
        </Title>

        <Stack>
                   
          {/* <StackItem> <ProjectDataFilterForm /></StackItem> */}
          <StackItem> <DemoProjectDataFilterForm /></StackItem>
        </Stack>
        

       
      </PageSection>
    );
  }
}
export { Dashboard };
