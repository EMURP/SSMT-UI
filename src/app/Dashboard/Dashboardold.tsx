// import React from 'react';

import { PageSection, Title } from '@patternfly/react-core';
import { Card,  CardBody } from '@patternfly/react-core';

import { SimpleInputGroups } from '@app/DateComponent/DateComponent';

import {DropdownComponent} from '../Dropdown/DropdownComponent';

import {SimpleTable} from '../myTable/SimpleTable';


import { TimesIcon, PlusCircleIcon } from '@patternfly/react-icons';

import {
  Button,
  InputGroup,
  TextInput,
  Dropdown,
  DropdownToggle,
  DropdownItem
} from '@patternfly/react-core';

import { Fetchdata } from '@app/fetchdata/Fetchdatatemp';


// import  {ProjectInfoTableComponent}  from '../ProjectInfo/ProjectInfoTableComponent';

const horizontalFlex = {
    display: 'flex',
    flexDirection: 'row'
} as React.CSSProperties;

const padding= {

  marginRight: '30px'

} as React.CSSProperties;


const Dashboardold React.FunctionComponent<{}> = () => {

  const [startHrs,setStartHrs]=useState(0);
  const [startMins,setStartMins]=useState(0);
  const [endHrs,setEndHrs]=useState(0);
  const [endMins,setEndMins]=useState(0);


  return(<PageSection>
    <Title headingLevel="h1" size="lg">Dashboard Page Title</Title>
      <div style={horizontalFlex}>
        <div style={padding}>
          <Card>
            <CardBody>Total Node: 37</CardBody>
          </Card>
        </div>
        <div style={padding}>
          <Card>
            <CardBody>Total Pods: 68</CardBody>
          </Card>
        </div>
        <div>
          <Card>
            <CardBody> Total Projects: 12</CardBody>
          </Card>
        </div>
      </div>

      <div>
        <h1>Active Time </h1>
      </div>


      <div style={horizontalFlex}>

        <div style = {padding}>
          <Card>
            <CardBody> Start Time From: </CardBody>
          </Card>

          <SimpleInputGroups/> 
          
       
          {/* <DropdownComponent setHrs={setStartHrs} setMins={setStartMins}/>         
          {startHrs}
          {startMins}  */}

        </div>

        <div style = {padding}>
          <Card>
            <CardBody> End Time To: </CardBody>
          </Card>

          <SimpleInputGroups/> 
{/* 
         <DropdownComponent setHrs={setEndHrs} setMins={setEndMins}/>
         {endHrs}
         {endMins} */}
          
            
         </div>

         <div>

         <Button isBlock>Search</Button>;
          
          </div>

      
      </div>
      
      <div>

      <h2 >Data from Fetch Data </h2>

        <Fetchdata/>

       
      </div> 

      {/* <div>
        <ProjectInfoTableComponent/>
      </div> */}

      
     



  </PageSection>)
}

export { Dashboardold};
