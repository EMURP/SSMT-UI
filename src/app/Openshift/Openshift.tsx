import * as React from 'react';
import { PageSection, Title }  from '@patternfly/react-core';
import {SimpleTable} from '@app/myTable/SimpleTable'
import { ChartPie, ChartThemeColor } from '@patternfly/react-charts';

//import {Table}from 'react-bootstrap-table';
//import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';

import { Card,  CardBody } from '@patternfly/react-core';
import {  } from '@patternfly/react-table';
import { Fetchdata } from '@app/fetchdata/fetchdata';

const horizontalFlex= {
  display: "flex",
  flexDirection: "row",
  marginRight: "30px",
  marginLeft: "30px"
} as React.CSSProperties

const padding= {

marginBottom: "30px",
marginRight: "30px",
marginLeft: "30px"

} as React.CSSProperties

const paddingChart= {

  marginBottom: "30px",
  marginRight: "30px",
  marginLeft: "30px",
  height: '275px',
  width: '300px'

} as React.CSSProperties

const Openshift: React.FunctionComponent<{}> = () => (


  <PageSection>


    <Title headingLevel="h1" size="lg">Openshift </Title>

  <div style={horizontalFlex}>

      <div style={padding}>
  
          <Card>
            <CardBody>Node Info Table </CardBody>
          </Card>

          <SimpleTable/>

      </div>

      <div style={padding}>

        <Card>
        <CardBody> Pod Info Table  </CardBody>
        </Card>

        <SimpleTable/>

    </div>
  
  </div>

  <div style={horizontalFlex}>

      <div style={padding}>
  
          <Card>
            <CardBody>Project Info Table </CardBody>
          </Card>

          <SimpleTable/>

      </div>

      <div style={paddingChart}>

     
      <ChartPie
      ariaDesc="Average number of Pods"
      ariaTitle="Pie chart example"
      constrainToVisibleArea={true}
      data={[{ x: 'Running', y: 35 }, { x: 'Failed', y: 55 }, { x: 'Terminated', y: 10 }]}
      height={275}
      labels={({ datum }) => `${datum.x}: ${datum.y}`}
      legendData={[{ name: 'Running: 35' }, { name: 'Failed: 55' }, { name: 'Terminated: 10' }]}
      legendPosition="bottom"
      padding={{
        bottom: 65,
        left: 20,
        right: 20,
        top: 20
      }}
      themeColor={ChartThemeColor.multiOrdered}
      width={400}
    />

    </div>
  
  </div>


  <div>
  <Fetchdata/>
  </div> 


  </PageSection>
)

export { Openshift };