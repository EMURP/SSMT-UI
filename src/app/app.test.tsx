import * as React from 'react';
import { App } from '@app/index';
import { Dashboard } from '@app/Dashboard/Dashboard';
import { Openshift } from '@app/Openshift/Openshift';
import { mount, shallow } from 'enzyme';
import { Button } from '@patternfly/react-core';
import { ChartPie, ChartThemeColor } from '@patternfly/react-charts';

describe('App tests', () => {
  test('should render default App component', () => {
    const view = shallow(<App />);
    expect(view).toMatchSnapshot();
  });

  it('should render a nav-toggle button', () => {
    const wrapper = mount(<App />);
    const button = wrapper.find(Button);
    expect(button.exists()).toBe(true);
  });

  it('should hide the sidebar when clicking the nav-toggle button', () => {
    const wrapper = mount(<App />);
    const button = wrapper.find('#nav-toggle').hostNodes();
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-expanded')).toBeTruthy();
    
    button.simulate('click');
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-collapsed')).toBeTruthy();
    
    
    expect(wrapper.find('#page-sidebar').hasClass('pf-m-exapnaded')).toBeFalsy();
  });

  it('should render the dashboard page', () => {
    const view = shallow(<Dashboard />);
    expect(view).toMatchSnapshot();
   
  });

  it('should render the openshift page', () => {
    const view = shallow(<Openshift />);
    expect(view).toMatchSnapshot();
   
  });

  it('should render the pie chart on openshift page', () => {
    const wrapper = mount(<Openshift />);
    
    const pie_chart = wrapper.find(ChartPie);
    expect(pie_chart.exists()).toBe(true);
   
  });

  

});
