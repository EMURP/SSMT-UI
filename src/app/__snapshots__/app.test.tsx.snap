// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<BrowserRouter>
  <AppLayout>
    <AppRoutes />
  </AppLayout>
</BrowserRouter>
`;

exports[`App tests should render the dashboard page 1`] = `
<Component>
  <Component
    headingLevel="h1"
    size="lg"
  >
    Dashboard Page Title
  </Component>
</Component>
`;
